File: 071__Simplify_Path.py
class Solution(object):
    def simplifyPath(self, path):
        """
        :type path: str
        :rtype: str
        """
        stack = []
        path = path.split("/")
        for p in path:
            if len(p) == 0:
                continue

            if p == ".":
                continue
            elif p == "..":
                if len(stack) != 0:
                    stack.pop(-1)
            else:
                stack.append(p)
        return "/" + "/".join(stack)
        

File: Java/071__Simplify_Path.java
class Solution {
    public String simplifyPath(String path) {
        path.replace("//", "/");
        String[] path_arr = path.split("/");
        Stack<String> stack = new Stack<>();
        for (String str : path_arr) {
            if (str.length() == 0) continue;
            if (str.equals("..")) {
                if (!stack.isEmpty()) stack.pop();
            } else if (!str.equals(".")) {
                stack.push(str);
            }
        }
        String result = "";
        while (!stack.isEmpty()) {
            result = "/" + stack.pop() + result;
        }
        if (result.length() == 0) result += "/";
        return result;
    }
}
